govcd/vapp_network.go

```diff
@@ -175,9 +176,13 @@ func (vapp *VApp) UpdateNetworkNatRulesAsync(networkId string, natRules []*types
 	}
 	networkToUpdate.Xmlns = types.XMLNamespaceVCloud
 
-	if networkToUpdate.Configuration.Features.NatService == nil {
+	if networkToUpdate.Configuration.Features.NatService == nil && networkToUpdate.Configuration.Features.FirewallService == nil {
 		return Task{}, fmt.Errorf("provided network isn't connected to org network or isn't fenced")
 	}
+	if networkToUpdate.Configuration.Features.NatService == nil {
+		networkToUpdate.Configuration.Features.NatService = &types.NatService{}
```

Comment:
Is it possible for the FirewallService to be nil at this point, do we need a similar check for it? If not, this asks for a comment in the code about why.

Reply:
this part already changed, please check latest changes.